### ISSUE TITLE
lookup subexpression inside a WITH helper not working

### ISSUE THREAD

**Issue by jaime3e** [zordius/lightncandy#252]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// The Template:
$template = <<<VAREND
	{{#each @root.indexes}}
		{{#with (lookup @root.objects this)}}
			{{id}}
		{{/with}}
	{{/each}}
VAREND;

// Helpers:
$helpers = array(
  'foo' => function () {
    return 'Hello World';
  },
);

// Partials:
$partials = array(
    'bar' => 'Yes, partial',
);

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_ERROR_EXCEPTION | LightnCandy::FLAG_HANDLEBARSJS_FULL,
  'helpers' => $helpers,
  'partials' => $partials,
));

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data= array( 'indexes'=>array('1','2'), 'objects'=>array('1'=>array('id'=>'111111'),'2'=>array('id'=>'2222222')));

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

Considering that my data has an array of indexes and then I have a big associative array with the objects and the id as the key, when I'm trying to get the full object with the index using WITH with a LOOKUP helper as a subexpression it doesn't work correctly. Please advice!
the data structure cannot be changed, thanks!!

**Comment by zordius** [comment]
Example jsfiddle created: https://jsfiddle.net/xwLsw7hf/

**Comment by zordius** [comment]
confirmed as a bug, will fix later.

**Comment by zordius** [comment]
It should be fix in master branch, please try again, thanks.

**Comment by zordius** [comment]
released in 1.0.1
